plugins/woocommerce/includes/class-wc-payment-gateways.php

```diff
@@ -113,6 +118,168 @@ public function init() {
+		foreach ( $this->payment_gateways as $gateway ) {
+			$option_key = $gateway->get_option_key();
+			self::add_action(
+				'add_option_' . $option_key,
+				function( $option, $value ) use ( $gateway ) {
+					$this->payment_gateway_settings_option_changed( $gateway, $value, $option );
+				},
+				10,
+				2
+			);
+			self::add_action(
+				'update_option_' . $option_key,
+				function( $old_value, $value, $option ) use ( $gateway ) {
+					$this->payment_gateway_settings_option_changed( $gateway, $value, $option, $old_value );
+				},
+				10,
+				3
+			);
+		}
+	}
+
+	/**
+	 * Callback for when a gateway settings option was added or updated.
+	 *
+	 * @param WC_Payment_Gateway $gateway   The gateway for which the option was added or updated.
+	 * @param mixed              $value     New value.
+	 * @param string             $option    Option name.
+	 * @param mixed              $old_value Old value. `null` when called via add_option_ hook.
+	 * @since 8.5.0
+	 */
+	private function payment_gateway_settings_option_changed( $gateway, $value, $option, $old_value = null ) {
+		if ( ! $this->was_gateway_enabled( $value, $old_value ) ) {
+			return;
+		}
+
+		// This is a change to a payment gateway's settings and it was just enabled. Let's send an email to the admin.
+		// "untitled" shouldn't happen, but just in case.
+		$this->notify_admin_payment_gateway_enabled( $gateway );
+	}
+
+	/**
+	 * Email the site admin when a payment gateway has been enabled.
+	 *
+	 * @param WC_Payment_Gateway $gateway The gateway that was enabled.
+	 * @return bool Whether the email was sent or not.
+	 * @since 8.5.0
+	 */
+	private function notify_admin_payment_gateway_enabled( $gateway ) {
+		$admin_email          = get_option( 'admin_email' );
+		$user                 = get_user_by( 'email', $admin_email );
+		$username             = $user ? $user->user_login : $admin_email;
+		$gateway_title        = $gateway->get_title();
+		$gateway_settings_url = esc_url_raw( self_admin_url( 'admin.php?page=wc-settings&tab=checkout&section=' . $gateway->id ) );
+		$site_name            = wp_specialchars_decode( get_option( 'blogname' ), ENT_QUOTES );
+		$site_url             = home_url();
+		/**
+		 * Allows adding to the addresses that receive payment gateway enabled notifications.
+		 *
+		 * @param array              $email_addresses The array of email addresses to notify.
+		 * @param WC_Payment_Gateway $gateway The gateway that was enabled.
+		 * @return array             The augmented array of email addresses to notify.
+		 * @since 8.5.0
+		 */
+		$email_addresses   = apply_filters( 'wc_payment_gateway_enabled_notification_email_addresses', array(), $gateway );
+		$email_addresses   = array_unique(
+			array_filter(
+				$email_addresses,
+				function( $email_address ) {
+					return filter_var( $email_address, FILTER_VALIDATE_EMAIL );
+				}
+			)
+		);
+		$email_addresses[] = $admin_email;
```

Comment:
```suggestion
		$email_addresses   = apply_filters( 'wc_payment_gateway_enabled_notification_email_addresses', array(), $gateway );
		$email_addresses   = array_filter(
			$email_addresses,
			function( $email_address ) {
				return filter_var( $email_address, FILTER_VALIDATE_EMAIL );
			}
		);
		$email_addresses[] = $admin_email;
		$email_addresses   = array_unique( $email_addresses );
```

:point_up: What do you think of moving `array_unique` a little further down? I'm thinking here that, despite the explanation in the docblock, a developer might also add the primary admin email address via their callback, in which case it would be included twice (which I assume mostly wouldn't be problematic, but one never knows).

Reply:
Whoops! That is actually exactly what I had intended, thanks for catching.